onnxruntime/python/tools/kernel_explorer/kernels/gemm_test.py

```diff
@@ -0,0 +1,100 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+
+import sys
+from itertools import product
+
+import kernel_explorer as ke
+import numpy as np
+import pytest
+
+
+def dtype_to_bytes(dtype):
+    type_map = {
+        "float16": 2,
+        "float32": 4,
+    }
+    return type_map[dtype]
+
+
+def dtype_to_funcs(dtype):
+    type_map = {
+        "float16": list(filter(lambda x: "RocblasGemm_half" in x, dir(ke))),
+        "float32": list(filter(lambda x: "RocblasGemm_float" in x, dir(ke))),
+    }
+    return type_map[dtype]
+
+
+def _test_gemm(size, dtype, func):
+    m, k, n = size
+
+    np.random.seed(0)
+    a = (np.random.rand(m, k).astype(dtype) - 0.5) * 2
+    b = (np.random.rand(k, n).astype(dtype) - 0.5) * 2
+    c = np.zeros((m, n)).astype(dtype)
+
+    dev_a = ke.DeviceArray(a)
+    dev_b = ke.DeviceArray(b)
+    dev_c = ke.DeviceArray(c)
+
+    f = getattr(ke, func)
+    va = f(ke.blas_op.N, ke.blas_op.N, n, m, k, 1.0, dev_b, n, dev_a, k, 0.0, dev_c, n)
+    va.Run()
+
+    dev_c.UpdateHostNumpyArray()
+
+    ref_c = a @ b
+
+    rtol = 1e-3 if dtype == "float16" else 1e-5
+    atol = 1e-3 if dtype == "float16" else 1e-5
+    np.testing.assert_allclose(ref_c, c, rtol=rtol, atol=atol)
+
+
+@pytest.mark.parametrize(
+    "size",
+    # product([5, 9], repeat=3)
+    product([1, 3, 4, 16, 124, 125, 126, 127, 128, 129, 130, 131, 132], repeat=3),
+)
+def test_gemm_all_sizes(size):
```

Comment:
Can we test for all transpose settings?

Reply:
Done. But removed some sizes to avoid combinatorial explosion.